executor/grant.go

```diff
@@ -554,7 +554,7 @@ func (e *GrantExec) grantTableLevel(priv *ast.PrivElem, user *ast.UserSpec, inte
 	if err != nil {
 		return err
 	}
-	sqlexec.MustFormatSQL(sql, " WHERE User=%? AND Host=%? AND DB=%? AND Table_name=%?", user.User.Username, user.User.Hostname, dbName, tblName)
+	sqlexec.MustFormatSQL(sql, " WHERE User=%? AND Host=%? AND lower(DB)=%? AND lower(Table_name)=%?", user.User.Username, user.User.Hostname, strings.ToLower(dbName), strings.ToLower(tblName))
```

Comment:
Do you want to use `lower` to handle existing data? That will cause table `mysql.tables_priv` can not be accessed by index

https://github.com/pingcap/tidb/blob/8d91a142503da041a80d8d21c877590a43f0db54/session/bootstrap.go#L147

Maybe we should normalize all data to lowercase during an upgrading? @tiancaiamao

Reply:
Thks very much. Yes that's a problem, let me think more. What I worry about is: if we make all data stored lowercase, will make the promption not same with table/db creators initial?

Reply:
can we create index wiith lower(db)...?

Reply:
> can we create index wiith lower(db)...?

Yes, let me clarify it. From my view we now have two problems:

1. There may have both uppercase and lowercase data for one privilege record. For example, user have created a SELECT privilege on db.TBL, and UPDATE privieleg on db.tbl. We should know what's the current behaviour (user will see SELECT or UPDATE or both?), and after your PR this behaviour is not changed.
2. Since the privilege is stored in a table, we should make use of primary key or index to access the table.